vizro-core/docs/pages/development/contributing.md

```diff
@@ -145,6 +145,21 @@ Vizro's version is given by `__version__` in [`src/vizro/__init__.py`](https://g
 
 To build the source distribution and wheel, run `hatch build`.
 
+### Steps to create a Vizro release
+
+The following steps are required to create a release of Vizro. Note that this is only done by maintainers:
+
+1. Checkout a new branch with a suitable name, e.g. `git checkout -b release/vx.y.z` (note that the branch name does not influence the versioning!)
+2. Run `hatch run prep-release` from the root folder of the package you want to release
+3. Check that the resulting `CHANGELOG.md` and `version` are as expected
+4. Commit all changes, raise a PR, and get approval from at least 2 maintainers
+5. After the PR is merged, also merge the automatically created version bump PR
+6. Verify that the newly released package on PyPI works as intended
+
+!!!note
+
+    If there are any changes to `main` after the release PR has been created and `hatch run prep-release` has been run, please restart the process as this may mess up the `CHANGELOG`.
```

Comment:
Maybe the merging queue could be useful due to this potential conflicts? https://github.blog/2023-07-12-github-merge-queue-is-generally-available/

Just an idea, definitely no need to pivot our current approach now

Reply:
Good shout, I'll take a closer look. Thanks for sharing